src/pcm-sensor-server.cpp

```diff
@@ -3788,6 +3790,14 @@ int mainThrows(int argc, char * argv[]) {
         for ( int i=1; i < argc; ++i ) {
             if ( check_argument_equals( argv[i], {"-d"} ) )
                 daemonMode = true;
+            else if ( check_argument_equals( argv[i], {"-l"} ) )
+            {
+                    if ( (++i) < argc ) {
+                        ip6Address = argv[i];
```

Comment:
Still no validation of the user input, see previous comment for how you could do this

Reply:
the comment about user input validation seems to be missing?

Reply:
I wrote a comment 2 days ago but I guess it never published. I wold like you to do some validation on the string you get from the user to make sure it is valid before trying to pass it around and then error out.

The URL class has some code to do IPv4 and IPv6 validation, you could take the code from that, put it in some static functions or refactor an IP class out of that and use that.